pkg/update/update_test.go

```diff
@@ -150,6 +153,44 @@ func TestGetAvailableUpdates(t *testing.T) {
 			want:    []types.AvailableUpdate{},
 			wantErr: true,
 		},
+		{
+			name: "uses installed channel id when multi-channel present",
+			args: args{
+				kotsStore: mockStore,
+				app: &apptypes.App{
+					ID:                "app-id",
+					SelectedChannelID: "channel-id2", // explicitly using the non-default channel
+				},
+				license: &kotsv1beta1.License{
+					Spec: kotsv1beta1.LicenseSpec{
+						ChannelID:   "channel-id",
+						ChannelName: "channel-name",
+						AppSlug:     "app-slug",
+						LicenseID:   "license-id",
+						Channels: []kotsv1beta1.Channel{
+							{
+								ChannelID:   "channel-id",
+								ChannelName: "channel-name",
+								IsDefault:   true,
+							},
+							{
+								ChannelID:   "channel-id2",
+								ChannelName: "channel-name2",
+								IsDefault:   false,
+							},
+						},
+					},
+				},
+			},
+			channelReleases: []upstream.ChannelRelease{},
+			setup: func(t *testing.T, args args, licenseEndpoint string) {
+				t.Setenv("USE_MOCK_REPORTING", "1")
+				args.license.Spec.Endpoint = licenseEndpoint
+				mockStore.EXPECT().GetCurrentUpdateCursor(args.app.ID, args.license.Spec.Channels[1].ChannelID).Return("1", nil)
+			},
+			want:    []types.AvailableUpdate{},
```

Comment:
can you please update the `newMockServerWithReleases` below to respect `selectedChannelID` query param similar to replicated app, and then validate that the available updates are returned for the correct channel?

Reply:
👍 yep - also updated the multi chan test to actually do some work.